### ISSUE TITLE
[BUG] initialTopMostItemIndex has no way to specify the alignment of the item

### ISSUE THREAD

**Issue by pranas** [petyosi/react-virtuoso#465]
**Describe the bug**
`followOutput` feature breaks when you have items that are taller than the container (ie 400px height item in a 300px container) in the initial list. It will scroll to the top of the last item and will not follow the updates. Once you manually scroll to the bottom of the last item and append more large items it will follow correctly. 

**Reproduction**
The issue can be reproduced using `follow-output.tsx` e2e test case with the following patch:

```
@@ -9,7 +9,7 @@ export default function App() {
   const [itemHeights, setItemHeights] = useState<Record<number, number>>({})
   const itemContent = useCallback(
     (index: number) => {
-      const height = itemHeights[index] ?? (index % 2 ? 30 : 40)
+      const height = itemHeights[index] ?? (index % 2 ? 400 : 500)
       return (
         <div style={{ height, background: 'white' }}>
           Item {index}
```

**To Reproduce**
0. Apply provided patch
1. Go to 'http://localhost:1234/follow-output'
2. Click on 'Append items'
4. List will not follow the output like it's supposed to

**Expected behavior**
It should scroll to the bottom of the last item in the list at 1st render and then correctly follow the updates.

**Screenshots**

https://user-images.githubusercontent.com/91422/136373330-fbf2b618-fe09-4ae4-b56a-b67fc4e90e04.mov

**Desktop (please complete the following information):**
 - OS: macOS 11.6 
 - Chrome 94

**Additional context**

**Comment by pranas** [comment]
Hi @petyosi, did you get a chance to make some progress on this?

**Comment by petyosi** [comment]
Sorry, no. If you have the capability, please submit a PR. It's not a big or complex task, but my schedule is awful.

**Comment by petyosi** [comment]
@WenWenBo Are you willing to sponsor or contribute?

**Comment by omgoshjosh** [comment]
i went for it with a very brief attempt and ran into runtime errors i don't have time to deal with but here's what i changed that seemed "downright trivial"
```typescript
// old
const initialTopMostItemIndex = u.statefulStream(0)
// new
const initialTopMostItemIndex = u.statefulStream({ index: 0, align: 'bottom' })

// snip

// old
u.filter(([_, index]) => index !== 0)
// new - unpack the index
u.filter(([_, { index }]) => index !== 0)
```

if anyone wants to take a stab at it, that does really seem to be all you need. looking at the scrollToIndex system it can take a number ("old") as well as the IndexWithAlign config object ("new") so i anticipate that my changes are sufficient but i just don't have time to debug for my local use case.

```typescript
export type IndexLocation = number | IndexLocationWithAlign

// snip

const scrollToIndex = u.stream<IndexLocation>()
```

**Comment by petyosi** [comment]
Solved in v2.8.0 thanks to @BigTomM .
